src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,587 @@
+//!
+//!     log_writer.write_comment(
+//!         "You can have comments anywhere in the log").unwrap();
+//!
+//!     log_writer
+//!         .write_histogram(
+//!             &h,
+//!             1.234,
+//!             5.678,
+//!             interval_log::Tag::new("im-a-tag"),
+//!             1.0)
+//!         .unwrap();
+//! }
+//!
+//! // `buf` is now full of stuff; we check for the first line
+//! assert_eq!("#Comments are great\n", &str::from_utf8(&buf).unwrap()[0..20]);
+//! ```
+
+extern crate base64;
+
+use std::{fmt, io, ops, str};
+use std::fmt::Write;
+
+use nom::{double, line_ending, not_line_ending, IResult};
+
+use super::super::{Counter, Histogram};
+use super::Serializer;
+
+/// Write headers for an interval log.
+///
+/// This type only allows writing comments and headers. Once you're done writing those things, use
+/// `into_log_writer()` to convert this into an `IntervalLogWriter`.
+pub struct IntervalLogHeaderWriter<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> {
+    internal_writer: InternalLogWriter<'a, 'b, W, S>,
+}
+
+impl<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> IntervalLogHeaderWriter<'a, 'b, W, S> {
+    /// Create a new log writer that writes to `writer` and serializes histograms with `serializer`.
+    pub fn new(writer: &'a mut W, serializer: &'b mut S) -> IntervalLogHeaderWriter<'a, 'b, W, S> {
+        IntervalLogHeaderWriter {
+            internal_writer: InternalLogWriter {
+                writer,
+                serializer,
+                text_buf: String::new(),
+                serialize_buf: Vec::new(),
+            },
+        }
+    }
+
+    /// Add a comment line.
+    ///
+    /// If you do silly things like write a comment with a newline character, you'll end up with
+    /// an un-parseable log file. Don't do that.
+    pub fn write_comment(&mut self, s: &str) -> io::Result<()> {
+        self.internal_writer.write_comment(s)
+    }
+
+    /// Write a StartTime log line. See the module-level documentation for more info.
+    ///
+    /// This should only be called once to avoid creating a confusing interval log.
+    pub fn write_start_time(&mut self, seconds_since_epoch: f64) -> io::Result<()> {
+        self.internal_writer.write_fmt(format_args!(
+            "#[StartTime: {:.3} (seconds since epoch)]\n",
+            seconds_since_epoch
+        ))
+    }
+
+    /// Write a BaseTime log line. See the module-level documentation for more info.
+    ///
+    /// This should only be called once to avoid creating a confusing interval log.
+    pub fn write_base_time(&mut self, seconds_since_epoch: f64) -> io::Result<()> {
+        self.internal_writer.write_fmt(format_args!(
+            "#[BaseTime: {:.3} (seconds since epoch)]\n",
+            seconds_since_epoch
+        ))
+    }
+
+    /// Once you're finished with headers, convert this into a log writer so you can write interval
+    /// histograms.
+    pub fn into_log_writer(self) -> IntervalLogWriter<'a, 'b, W, S> {
```

Comment:
See notes above about making this more builder-like, and then only committing to a writer + serializer at this point (rather than when creating the header writer).